docs/Displaying a User Interface/Alerts/index.md

```diff
@@ -8,128 +8,205 @@ The alert will persist on the screen until the timeout has elapsed, or the user
-// Not all head units support the progress indicator
-alert.progressIndicator = @YES;
+SDLSoftButton *button1 = [[SDLSoftButton alloc] init];
+button1.text = @"<# Button Text #>";
+button1.type = SDLSoftButtonTypeText;
+button1.softButtonID  = @<#Soft Button Id#>;
+button1.handler = ^(SDLOnButtonPress *_Nullable buttonPress,  SDLOnButtonEvent *_Nullable buttonEvent) {
+if (buttonPress == nil) {
+    return;
+}
+// create a custom action for the selected button
+};
 
-// Text-to-speech
-alert.ttsChunks = [SDLTTSChunk textChunksFromString:@"<#Text to speak#>"];
+SDLSoftButton *button2 = [[SDLSoftButton alloc] init];
+button2.text = @"<# Button Text 2 #>";
+button2.type = SDLSoftButtonTypeText;
+button2.softButtonID  = @<#Soft Button Id#>;
+button2.handler = ^(SDLOnButtonPress *_Nullable buttonPress,  SDLOnButtonEvent *_Nullable buttonEvent) {
+if (buttonPress == nil) {
+    return;
+}
+// create a custom action for the selected button
+};
 
-// Special tone played before the tts is spoken
-alert.playTone = @YES;
+[softButtons addObject: button1];
+[softButtons addObject: button2];
 
-// Soft buttons
-SDLSoftButton *okButton = [[SDLSoftButton alloc] init];
-okButton.text = @"OK";
-okButton.type = SDLSoftButtonTypeText;
-okButton.softButtonID = @<#Soft Button Id#>;
-okButton.handler = ^(SDLOnButtonPress *_Nullable buttonPress,  SDLOnButtonEvent *_Nullable buttonEvent) {
-    if (buttonPress == nil) {
-      return;
-    }
-
-    // create a custom action for the selected button
-};
+alert.softButtons = softButtons;
+```
 
-alert.softButtons = @[okButton];
+#### Swift
+```swift
+let alert = SDLAlert(alertText1: "<#Line 1#>", alertText2: "<#Line 2#>", alertText3: "<#Line 3#>")
 
-// Send the alert
-[self.sdlManager sendRequest:alert withResponseHandler:^(SDLRPCRequest *request, SDLRPCResponse *response, NSError *error) {
-    if ([response.resultCode isEqualToEnum:SDLResultSuccess]) {
-      // alert was dismissed successfully
-    }
-}];
+var softButtons = [SDLSoftButton]()
+
+var button1 = SDLSoftButton()
+button1.text = "<# Button Text #>"
+button1.type = .text
+button1.handler = { buttonPress, buttonEvent in
+    guard buttonPress != nil else { return }
+    <# Create a custom action for the selected button #>
+}
+
+var button2 = SDLSoftButton()
+button2.text = "<# Button Text 2 #>"
+button2.type = .text
+button2.handler = { buttonPress, buttonEvent in
+    guard buttonPress != nil else { return }
+    <# Create a custom action for the selected button #>
+}
+
+softButtons.append(button1)
+softButtons.append(button2)
+
+alert.softButtons = softButtons;
+```
+!@
+
+@![android,javaSE,javaEE]
+` TODO - code example `
+!@
+
+## Additonal Alert Paramaters
+@![iOS]
+
+### Timeouts
+An optional timeout can be added that will dimiss the alert when the duration is over.  Typical timeouts are between 3 and 5 seconds. If omitted a default of 5 second is used.
+#### Objective-C
+```objc
+alert.duration = @(4000);
```

Comment:
Add a comment that this is in milliseconds

Reply:
added